### DISCUSSION TITLE
Testing components and rendered routes?

### DISCUSSION THREAD

**Discussion by dts** [nuxt/nuxt#16980] [Questions]
In projects I've built in Vue3+Vite, there's a testing pattern that I use (with `@testing-library/vue`) that looks something like this:

```
const { findByText } = renderApp('/settings/user');
```

Which behind the scenes, sets up plugins, mounts routes etc. I was expecting to be able to do something similar with Nuxt - i.e. yank the vue & vite plugins that make up Nuxt (auto-imports, routing, etc) and shove them into `@testing-library/vue` & `vitest` - allowing me to get at the contents of a rendered route as a testable set of @testing-library matches, and run them through their paces.  I assume such a thing is possible, but there were enough plugins that I didn't want to go one by one.  I could get individual components to work using `mount()`, but without AutoImports etc, there's a pretty low ceiling on what I can actually test.

I'm so excited by the work that's happening!

**Comment by danielroe** [comment]
You may want to follow https://github.com/nuxt/framework/issues/6496.

**Reply by dts** [reply]
Ah! That's super informative! Playing around with it, I can get at quite a bit of good stuff that is close to what I want (e.g. component lists, lists & implementations of plugins that are going to be loaded into Vue).  It's possible I'd be able to plug in to these things to get what I want, but I suspect it would be quite brittle. Building `renderComponent()` and `renderApp()` into `@nuxt/test-utils` I think is probably the thing I should look into doing.

**Reply by dts** [reply]
Though it looks like I may be able to get something close with `createPage()` and playwright matchers... though then I'd need to figure out a way to get coverage information back out of the browser.
